### PR TITLE
Allow actions to work on fork

### PR DESCRIPTION
**Pull request by AngelOnFira**
Part of RVT-3554

### CODE REVIEW

**Comment by AngelOnFira** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/371?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #360** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/360?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #357** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/357?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #358** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/358?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #359** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/359?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #365** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/365?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #366** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/366?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #371** <a href="https://app.graphite.dev/github/pr/rivet-gg/rivet/371?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rivet-gg/rivet/371?utm_source=stack-comment).

File: .github/actions/pre-init/action.yaml [hunk 1]
```diff
@@ -27,17 +27,29 @@ runs:
...
+    - name: Debug GITHUB_ENV
+      shell: bash -e {0}
+      run: |
+        whoami
+        # stat /home/runner
+        # mkdir -m 0755 /nix && chown runner /nix
+        # stat /home/runner
+        # echo "TEST=TEST" >> $GITHUB_ENV
+        # echo "GITHUB_ENV:"
+        # cat $GITHUB_ENV
```
Comment (NathanFlurry) [inline .github/actions/pre-init/action.yaml:52]: Can we remove this code?

File: .github/actions/pre-init/action.yaml [hunk 2]
```diff
@@ -27,17 +27,29 @@ runs:
...
-        echo 'SCCACHE_BUCKET=rivet-sccache' >> $GITHUB_ENV
-        echo 'SCCACHE_ENDPOINT=https://2a94c6a0ced8d35ea63cddc86c2681e7.r2.cloudflarestorage.com/rivet-sccache' >> $GITHUB_ENV
-        echo 'SCCACHE_REGION=auto' >> $GITHUB_ENV
-        echo 'AWS_SECRET_ACCESS_KEY=${{ inputs.sccache_aws_secret_access_key }}' >> $GITHUB_ENV
-        echo 'AWS_ACCESS_KEY_ID=${{ inputs.sccache_aws_access_key_id }}' >> $GITHUB_ENV
+        if [[ -n "${{ inputs.sccache_aws_secret_access_key }}" && -n "${{ inputs.sccache_aws_access_key_id }}" ]]; then
+          # Expose sccache config for all future calls of nix-shell, which configures sccache respectively
+          echo 'USE_SCCACHE=1' >> $GITHUB_ENV
+          echo 'SCCACHE_BUCKET=rivet-sccache' >> $GITHUB_ENV
+          echo 'SCCACHE_ENDPOINT=https://2a94c6a0ced8d35ea63cddc86c2681e7.r2.cloudflarestorage.com/rivet-sccache' >> $GITHUB_ENV
```
Comment (NathanFlurry) [inline .github/actions/pre-init/action.yaml:37]: Thoughts on throwing the SCCACHE_ENDPOINT in to a secret for eacy configuration, even though it isn't sensitive?

**Comment by AngelOnFira** [comment]
Closing this for now, probably good to look back into later.

**Pull request closed without merging**